src/main/java/net/snowflake/ingest/streaming/internal/ArrowRowBuffer.java

```diff
@@ -80,14 +80,10 @@ class ArrowRowBuffer extends AbstractRowBuffer<VectorSchemaRoot> {
   // Allocator used to allocate the buffers
   private final BufferAllocator allocator;
 
-  /**
-   * Construct a ArrowRowBuffer object
-   *
-   * @param channel client channel
-   */
-  ArrowRowBuffer(SnowflakeStreamingIngestChannelInternal<VectorSchemaRoot> channel) {
-    super(channel);
-    this.allocator = channel.getAllocator();
+  /** Construct a ArrowRowBuffer object. */
+  ArrowRowBuffer(BufferConfig bufferConfig) {
+    super(bufferConfig);
+    this.allocator = bufferConfig.allocator;
```

Comment:
Why we're saving the allocator in both the super class and here?

Reply:
I do not see it in super class here but actually we will need it there eventually for Parquet as well so I will move it there.